docs/design/2021-09-22-stale-read.md

```diff
@@ -0,0 +1,132 @@
+# Proposal: Support Stale Read in TiDB
+
+- Author(s): [Yisaer](https://github.com/Yisaer) (Song Gao)
+- Last updated: 2021-08-18
+- Tracking Issue: https://github.com/pingcap/tidb/issues/21094
+- Related Document: https://docs.google.com/document/d/1dSbXbudTK-hpz0vIBsmpfuzeQhhTFJ-zDenygt8fmCk/edit?usp=sharing
+
+## Introduction
+
+This proposal aims to support Stale Read and described what stale read looks like and how to keep compatibility with other
+query.
+
+## Motivation or Background
+
+Currently, TiDB support Follower Read to read data from follower replicas in order to increase the query throughput. TiDB
+also support Snapshot Read to read old data with given snapshot timestamp. However, Follower Read still require an rpc calling
+between follower replica and leader replica during querying which also increased query latency. For Snapshot Read, it only read
+data from leader replica which might be bottleneck especially in hot spot cases.
+
+Stale Read is more like the combination of Follower Read and Snapshot Read. It supports tidb to read data from both leader replicas
+and follower replicas with no extra rpc calling during followers and leader. It also supports read data with given snapshot timestamp
+like Snapshot Read.
+
+## Detailed Design
+
+### Stale Read Syntax Overview
+
+Here we will introduce how to use Stale Read in TiDB. 
+
+1. Use Stale Read with given specific timestamp in single statement:
+
+```sql
+SELECT * FROM t AS OF TIMESTAMP '2021-09-22 15:04:05' WHERE id = 1;
+SELECT * FROM t AS OF TIMESTAMP NOW(20) WHERE id = 1;
```

Comment:
`NOW(20)` looks like calling the [NOW function](https://dev.mysql.com/doc/refman/8.0/en/date-and-time-functions.html#function_now) with fsp=20.

Reply:
good cactch, updated.